### PR TITLE
Removes VirtualBox references throughout docs

### PR DESCRIPTION
**Pull request by conorsch**
## Status

Ready for review

## Description of Changes

Supplements https://github.com/freedomofpress/securedrop/pull/5922. Edits the dev docs to remove explicit mentions of Virtualbox, preferring Libvirt (or Qubes) for any of the VM docs. Mostly this affects the macOS docs. For macOS, we're now more clearly directing folks to use the container dev env.

## Testing
* Review and merge https://github.com/freedomofpress/securedrop-docs/pull/214 first! Then rebase this PR.
* Visual review. I recommend generating the docs locally from this branch and stepping through the changed pages. 

## Release 
No concerns, these changes are dev-only.

## Checklist (Optional)

- [ ] Doc linting (`make docs-lint`) passed locally
- [ ] Doc link linting (`make docs-linkcheck`) passed
- [ ] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

**Comment by conorsch** [comment]
Marking as blocked for now, since I based these changes on https://github.com/freedomofpress/securedrop-docs/pull/214

**Comment by eloquence** [comment]
(Can be rebased now that #214 is merged.)

File: docs/development/setup_development.rst [hunk 1]
```diff
@@ -8,12 +8,10 @@ Overview
 
 SecureDrop is a multi-machine design. To make development and testing
 easy, we provide a set of virtual environments, each tailored for a
-specific type of development task. We use Vagrant, VirtualBox, and
+specific type of development task. We use Vagrant, Molecule, and
```
Comment (creviera) [inline docs/development/setup_development.rst:11]: Do also want to include Libvirt in this list?

Comment (conorsch) [reply docs/development/setup_development.rst:11]: I think that's a fine idea. Do you mind appending as part of review, @creviera ?

**Review by creviera — APPROVED** [review]
lgtm - conor also reviewed my small commit

**Pull request merged**